Description: manually adding hardening flags
Forwarded: not-needed
--- a/cpdb/Makefile.am
+++ b/cpdb/Makefile.am
@@ -19,7 +19,7 @@
 libcpdb_la_LIBADD += $(GIO_LIBS)
 libcpdb_la_LIBADD += $(GIOUNIX_LIBS)
 
-libcpdb_la_LDFLAGS = -no-undefined -version-info 2
+libcpdb_la_LDFLAGS = -no-undefined -version-info 2 -Wl,-z,now -Wl,-z,relro
 
 
 libcpdb_frontend_la_SOURCES = cpdb-frontend.c \
@@ -35,7 +35,7 @@
 libcpdb_frontend_la_LIBADD += $(GIO_LIBS)
 libcpdb_frontend_la_LIBADD += $(GIOUNIX_LIBS)
 
-libcpdb_frontend_la_LDFLAGS = -no-undefined -version-info 2
+libcpdb_frontend_la_LDFLAGS = -no-undefined -version-info 2 -Wl,-z,now -Wl,-z,relro
 
 
 cpdb_headersdir = $(includedir)/cpdb
